Windows support

jacobtread opened the pull request:
**Description of the changes**
Please provide a clear and concise description of the changes you made.

Added support for windows using the GetExtendedTcpTable, GetExtendedUdpTable, OpenProcess, and TerminateProcess I used the windows-sys crate for implementing this

**Related issue(s)**
If this PR is related to an existing issue, please link to it using the `Fixes #issue_number` or `Closes #issue_number` syntax.

Closes #12 

**Type of change**
Please select one or multiple of the following options:
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Code cleanup (refactoring or improving code quality)
- [x] Documentation update (adding or updating documentation, updating README)

**Checklist:**
- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

**Additional information**
Add any other information or screenshots about the pull request here.

Sorry about the Cargo.toml assets formatting my editor refused to save without formatting it

jacobtread committed:
Added windows support

jacobtread committed:
Adjusted documentation to fit codestyle

jacobtread committed:
Added missing features for windows-sys and merged imports

jacobtread committed:
Updated README.md to include "and Windows"

jacobtread committed:
Added logging messages to match other platforms

jkfran commented:
This is incredible work. Thank you so much ❤️, I will give you a proper review soon.

jkfran commented:
It seems that killport it's returning a success message, but the process is not actually killed.

![image](https://github.com/jkfran/killport/assets/6353928/802c852c-9957-44df-bad6-ff16c667937f)

I also tried running nginx on port 80 as Admin and running killport as admin to kill it, but it didn't work.

jacobtread commented:
That's strange I'll have a look when I get the chance to see if I can find out why that's happening

jacobtread commented:
~~Oh okay I found the issue, looks like I need to write some extra code to give the current process the correct permissions to access other user processes. I will get to writing this extra code, ill also brush up the error handling to make it more clear of what actually went wrong.~~

~~Will bump this pull request once I've implemented the changes~~

jacobtread commented:
Actually I don't think that was the issue, from the looks of it nginx has two processes and this program only kills the first process (The one binding port) which immediately gets started again by the second process

As for that system http listener I'm unable to reproduce a successful message from that, for me I fail to OpenProcess  I think that process becomes protected by windows as its refusing to let me open a process handle to it. As for other apps I have no problem terminating them I've tested on a few different servers of mine and haven't had any issue. 

The initial null handle check I noticed was failing with a zero instead of a nullptr though so I've updated that to handle it correctly

Looks like these problems might be specific two these two specific servers?

![nginx](https://github.com/jkfran/killport/assets/33708767/14e31368-5b04-4144-aaca-8b6aae387c8a)

jacobtread committed:
Fixed null handle check

jacobtread committed:
Replaced PIDs collection, and used unsafe block

PIDs collection is now a HashSet to prevent duplicates causing errors trying to double terminate
Wrapped main calls in one big unsafe block to cleanup the code
Replaced killed variable with checks that ensure always a specific outcome

jkfran commented:
@jacobtread In the nginx case do you know if we could find the parent process and kill it as well?

jacobtread commented:
I've produced a working version which collects the parents processes and from my experience it seems as though its able to successfully kill nginx I will add this commit to the pull request shortly after finalizing the code for it

jacobtread committed:
Parent process collection and killing

jacobtread commented:
Changes are up 👍🏻

jkfran reviewed (approved):
@jacobtread It works very well. I had to use Windows for the first time in a long time to test it out :D 

It would be nice to add support for different signals as well. `kill` from Gnuwin32 supports the following signals:
```
INT
ILL
FPE
KILL
SEGV
TERM
CHLD
ABRT
```

But Windows support is already a big step. I will merge your changes and release a new version soon.

Thank you so much ❤️
